Spigot-API-Patches/0040-Add-EntityZapEvent-replace-PigZapEvent-include-villa.patch

```diff
@@ -0,0 +1,86 @@
+From 68f360558a1e027db3032bc3406f6d41af286ae9 Mon Sep 17 00:00:00 2001
+From: AlphaBlend <whizkid3000@hotmail.com>
+Date: Sun, 16 Oct 2016 23:19:34 -0700
+Subject: [PATCH] Add EntityZapEvent, replace PigZapEvent, include villagers
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/entity/EntityZapEvent.java b/src/main/java/com/destroystokyo/paper/event/entity/EntityZapEvent.java
+new file mode 100644
+index 0000000..d2f0d50
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/entity/EntityZapEvent.java
+@@ -0,0 +1,56 @@
++package com.destroystokyo.paper.event.entity;
++
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.LightningStrike;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++
++/**
++ *  Fired when lightning strikes an entity
++ */
++public class EntityZapEvent extends EntityEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++    private boolean cancelled;
++    private final LightningStrike bolt;
++    private final Entity replacedEntity;
++
++    public EntityZapEvent(final Entity entity, final LightningStrike bolt, final Entity replacedEntity) {
++        super(entity);
++        this.bolt = bolt;
++        this.replacedEntity = replacedEntity;
++    }
++
++    public boolean isCancelled() {
++        return cancelled;
++    }
++
++    public void setCancelled(boolean cancel) {
++        this.cancelled = cancel;
++    }
++
++    /**
++     * Gets the lightning bolt that is striking the entity.
++     * @return The lightning bolt responsible for this event
++     */
++    public LightningStrike getBolt() {
++        return bolt;
++    }
++
++    /**
++     * Gets the entity that will replace the struck entity.
++     * @return The entity that will replace the struck entity
++     */
++    public Entity getReplacedEntity() {
++        return replacedEntity;
++    }
++
++    @Override
++    public HandlerList getHandlers() {
++        return handlers;
++    }
++
++    public static HandlerList getHandlerList() {
++        return handlers;
++    }
++}
+diff --git a/src/main/java/org/bukkit/event/entity/PigZapEvent.java b/src/main/java/org/bukkit/event/entity/PigZapEvent.java
+index aa80ebf..3470d9d 100644
+--- a/src/main/java/org/bukkit/event/entity/PigZapEvent.java
++++ b/src/main/java/org/bukkit/event/entity/PigZapEvent.java
+@@ -15,6 +15,10 @@ public class PigZapEvent extends EntityEvent implements Cancellable {
```

Comment:
Yes, the correct way to handle this is to extend EntityZapEvent, remove the getHandlers and getHandlerList from this class (comment out)

Also, do not deprecate. It's a bukkit API that will never be removed, and can be useful for people who only want to listen to pig zaps.